src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -31,6 +31,7 @@ internal sealed class MethodCompiler : CSharpSymbolVisitor<TypeCompilationState,
         private readonly PEModuleBuilder _moduleBeingBuiltOpt; // Null if compiling for diagnostics
         private readonly Predicate<Symbol> _filterOpt;         // If not null, limit analysis to specific symbols
         private readonly DebugDocumentProvider _debugDocumentProvider;
+        private readonly MethodSymbol _entryPointOpt;
```

Comment:
>private readonly MethodSymbol _entryPointOpt; [](start = 8, length = 45)

Wouldn't it make sense to store the entry point on PEModuleBuilder to keep MethodCompiler smaller - all methods in the same module will have the same value here, right?

Reply:
Thanks. I'll try that

Reply:
This is trouble. We currently need the PEModuleBuilder to figure out the entry point (it's an input to `GetEntryPoint`).
I don't see a good way to economize more (I'd rather not mutate the PEModuleBuilder).

---
In reply to: [271426312](https://github.com/dotnet/roslyn/pull/34548#discussion_r271426312) [](ancestors = 271426312)

Reply:
Could it be `Lazy<MethodSymbol>` on `PEModuleBuilder`?

Reply:
I looked into this more. When building, we'll have one `PEModuleBuilder` and one `MethodCompiler`. Can you clarify why you are concerned about the size of `MethodCompiler` as opposed to that of `PEModuleBuilder`?

---
In reply to: [271823191](https://github.com/dotnet/roslyn/pull/34548#discussion_r271823191) [](ancestors = 271823191)

Reply:
I guess I was confused. I though we have one MethodCompiler per method. Never mind then. Now wondering why we need a MethodCompiler and ModuleBuilder to be separate types :).